CRM/Core/Payment.php

```diff
@@ -684,20 +684,25 @@ public static function handlePaymentMethod($method, $params = array()) {
     $method = 'handle' . $method;
     $extension_instance_found = FALSE;
 
-    // In all likelihood, we'll just end up with the one instance returned here. But it's
-    // possible we may get more. Hence, iterate through all instances ..
+    // In all likelihood, we'll just end up with the one instance
+    // returned here. But it's possible we may get more. Hence,
+    // iterate through all instances ..
 
     while ($dao->fetch()) {
-      // Check pp is extension - is this still required - surely the singleton below handles it.
+      // Check pp is extension - is this still required - surely the
+      // singleton below handles it.
       $ext = CRM_Extension_System::singleton()->getMapper();
       if ($ext->isExtensionKey($dao->class_name)) {
         $paymentClass = $ext->keyToClass($dao->class_name, 'payment');
         require_once $ext->classToPath($paymentClass);
       }
+      else {
+        $paymentClass = $dao->class_name;
```

Comment:
I don't quite understand this because I don't think this is the first core processor that can be loaded this way - unless the others don't actually work since I know we haven't been pro-actively pushing the url change.

My expectation is that this line

$processorInstance = Civi\Payment\System::singleton()->getById($dao->processor_id); 

should work here - and ideally also could replace the lines above.